raiden/tests/unit/transfer/test_utils.py

```diff
@@ -17,3 +27,73 @@
 )
 def test_hash_balance_data(values, expected):
     assert hash_balance_data(values[0], values[1], values[2]) == expected
+
+
+def test_events_for_onchain_secretreveal_with_unfit_channels():
+    settle = factories.TransactionExecutionStatusProperties()
+    settled = factories.create(factories.NettingChannelStateProperties(settle_transaction=settle))
+    secret = factories.UNIT_SECRET
+    block_hash = factories.make_block_hash()
+
+    events = events_for_onchain_secretreveal(settled, secret, 10, block_hash)
+    assert not events, "Secret reveal event should not be generated for settled channel"
+
+    settle = factories.replace(settle, result=TransactionExecutionStatus.FAILURE)
+    unusable = factories.create(factories.NettingChannelStateProperties(settle_transaction=settle))
+
+    events = events_for_onchain_secretreveal(unusable, secret, 10, block_hash)
+    assert not events, "Secret reveal event should not be generated for unusable channel."
+
+
+def test_events_for_onchain_secretreveal_typechecks_secret():
+    channel = factories.create(factories.NettingChannelStateProperties())
+    block_hash = factories.make_block_hash()
+    with pytest.raises(ValueError):
+        events_for_onchain_secretreveal(channel, "This is an invalid secret", 10, block_hash)
+
+
+def test_get_state_change_by_locksroot():
+    canonical_id = factories.UNIT_CANONICAL_ID
+    locksroot = factories.UNIT_SECRETHASH
+    sender = factories.UNIT_TRANSFER_SENDER
+
+    expected = {
+        "balance_proof.canonical_identifier.chain_identifier": str(canonical_id.chain_identifier),
+        "balance_proof.canonical_identifier.token_network_address": to_checksum_address(
+            canonical_id.token_network_address
+        ),
+        "balance_proof.canonical_identifier.channel_identifier": str(
+            canonical_id.channel_identifier
+        ),
+        "balance_proof.locksroot": encode_hex(locksroot),
+        "balance_proof.sender": to_checksum_address(sender),
+    }
+    storage = Mock()
+
+    def check(arg):
+        assert check_dict_nested_attrs(arg, expected)
+
+    storage.get_latest_state_change_by_data_field = Mock(side_effect=check)
```

Comment:
This is testing the Mock, not our real code. Please either remove it or use a real db.

Reply:
Fine, I'll rewrite the test using a real db then. There's no way to reach 90% module coverage without it.